pulp_ansible/app/downloaders.py

```diff
@@ -0,0 +1,74 @@
+from logging import getLogger
+import asyncio
+import backoff
+import json
+
+from aiohttp.client_exceptions import ClientResponseError
+
+from pulpcore.plugin.download import http_giveup, HttpDownloader
+
+
+log = getLogger(__name__)
+
+
+class TokenAuthHttpDownloader(HttpDownloader):
+    """
+    Custom Downloader that automatically handles Token Based and Basic Authentication.
+
+    Additionally, use custom headers from DeclarativeArtifact.extra_data['headers']
```

Comment:
Do these come from DeclarativeArtifact? I don't see them set anywhere.

Reply:
I copied from `pulp_container` and forgot to remove it